### ISSUE TITLE
[Feature] Get playlist track number

### ISSUE THREAD

**Issue by ScubaAlex-WPG** [exislow/tidal-dl-ng#165]
### Current Situation

{album_track_num} will return the track number for the album where that track is from.
Having said that, downloading a playlist will present lots of files with the same track number.

### Suggestion / Feature Request

create a {playlist_track_num} so we can download the whole playlist files and each track numbers will be unique, based on the playlist track number.
The same concept could be applied to Mixes.

![image](https://github.com/user-attachments/assets/e86bef2d-9731-4065-bed4-c97d4f00e7bf)

**Comment by frankwu8765** [comment]
Yes, I also think it's a very important feature for playlists!

**Comment by YuGo-git-hub** [comment]
I join with the same request. It is very important to respect the order of tracks in tracklists and mixes. I hope in the future there will be an option {playlist_track_num}.

**Comment by WinFan3672** [comment]
I know that a +1 comment is hardly useful, but I'm surprised this hasn't been implemented.

**Comment by thecurtis76** [comment]
I'm throwing my hat in on this one, too. This would be a very useful feature for playlists and mixes.

**Comment by mversion** [comment]
+1 here.

Using a tagger based on modified timestamp does the trick, but would be good if we didn't have to worry about it.

**Comment by SivaMarti** [comment]
+1  

Please add this feature. Thank You!

**Comment by athanas08** [comment]
+1 too.

**Comment by nilleiz** [comment]
+1 (sorry)

**Comment by Beking8** [comment]
+1

**Comment by MarkD1212** [comment]
Great program, I would really like to see this feature added.

**Comment by b1ggi** [comment]
Is it also possible to include this „playlist track num“ if downloaded from fav_tracks?

**Comment by djdk80** [comment]
+1

**Comment by NaNoXy** [comment]
+1 Very useful for people listening a lot to playlist, and want to keep the personalized order

**Comment by heartless-jeden** [comment]
@ to all those, who can bear with command line apps:

As much as I appreciate all the effort put into creating and maintaining tidal-dl-ng, I really wanted to have my tracks numbered just like in the original playlist.

FYI: _tiddl_ supports this, though it's lacking in many different aspects.

**Comment by rital06** [comment]
Yes, please: downloading a playlist without its sequence numbers is useless. A playlist is nice because of the exact concatenation of songs in that precise order!

**Comment by exislow** [comment]
This feature is now implemented. You could use something like `format_playlist: str = "Playlists/{playlist_name}/{list_pos}. {artist_name} - {track_title}"` to enable it.

Watch out: As soon as a list sequence changes or a track becomes unavailable in you playlist, all subsequent tracks get re-downloaded, even if `skip_existing` is enabled. Duplicate check happens only based on file names.

**Comment by ftsunny** [comment]
Wow, this is really the update I've been waiting for. Thank you! @exislow

**Comment by ScubaAlex-WPG** [comment]
Very much appreciated @exislow 
:)

**Comment by NorthGraves** [comment]
Thank you, works great!
